### PR TITLE
Fixes for linux and windows builds

### PR DESCRIPTION
**Pull request by Gustav-Simonsson**

### CODE REVIEW

**Comment by Gustav-Simonsson** [comment]
@LefterisJP ping

**Comment by LefterisJP** [comment]
Thanks for opening a PR on the upstream immediately. Looks fine except for the commented out code. It's never a permanent solution to just comment out code :)

File: ethashc.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package ethash
 
 /*
 #cgo CFLAGS: -std=gnu99 -Wall
+#cgo windows CFLAGS: -mno-stack-arg-probe
 #cgo LDFLAGS: -lm
```
Comment (fjl) [inline ethashc.go]: We really need to add a comment about this.

**Comment by Gustav-Simonsson** [comment]
@fjl @LefterisJP updated

**Comment by LefterisJP** [comment]
Still not what I meant. Please leave the #ifdefs as they were before you touched that code. Don't touch the code of that file at all.

This conditional compilation flag is needed for MSVC which you do not use. Instead remove `utils_win32.c` from compiling as I suggested in my previous comment.

**Comment by Gustav-Simonsson** [comment]
@LefterisJP updated again, please review

**Comment by LefterisJP** [comment]
Hmm ... not yet done. Undo all your changes in `util.h`

**Pull request merged**